Fix byte-compile warnings in Emacs 29.
Drop cl-lib which was needed only for cl-letf.

--- a/pinentry.el
+++ b/pinentry.el
@@ -49,8 +49,6 @@
 
 ;;; Code:
 
-(eval-when-compile (require 'cl-lib))
-
 (defgroup pinentry nil
   "The Pinentry server"
   :version "25.1"
@@ -92,7 +90,7 @@
 
 ;; These error codes are defined in libgpg-error/src/err-codes.h.in.
 (defmacro pinentry--error-code (code)
-  (logior (lsh 5 24) code))
+  (logior (ash 5 24) code))
 (defconst pinentry--error-not-implemented
   (cons (pinentry--error-code 69) "not implemented"))
 (defconst pinentry--error-cancelled
@@ -174,7 +172,7 @@
       (ignore-errors
         (let (delete-by-moving-to-trash)
           (delete-file server-file)))
-      (cl-letf (((default-file-modes) ?\700))
+      (with-file-modes ?\700
         (setq pinentry--server-process
               (make-network-process
                :name "pinentry"
